src/components/capture-area/BeforeConstraintRegex.vue

```diff
@@ -0,0 +1,117 @@
+        class="rounded"
+        size="x-small"
+        variant="text"
+    >
+        {{ !expand ? 'More Options' : 'Less Options' }}
+    </v-btn>
+
+    <v-expand-transition>
+        <div v-if="expand" class="pa-4">
+            <v-combobox
+                v-model="currentMatchingOption"
+                label="Matching"
+                :items="matchingOptions"
+                variant="underlined"
+            ></v-combobox>
+
+            <v-combobox
+                v-model="currentSlicingOption"
+                label="Slicing"
+                :items="slicingOptions"
+                variant="underlined"
+            ></v-combobox>
+
+            <v-combobox
+                v-model="currentSimilarityOption"
+                label="Similarity"
+                :items="similarityOptions"
+                variant="underlined"
+            ></v-combobox>
+
+            <v-text-field
+                v-if="currentMatchingOption === 'Approximate'"
+                v-model="currentNumberOfMatches"
+                label="Number of Matches"
+                placeholder="Enter a number"
+                variant="outlined"
+                clear-icon="mdi-close-circle"
+                :rules="[(v:number) => !!v || 'Please enter a number']"
+                hide-details="auto"
+            >
+                <template v-slot:prepend-inner>
+                    <v-icon icon="mdi-numeric"></v-icon>
+                </template>
+            </v-text-field>
+
+            <v-btn
+                @click="reset()"
+                class="rounded mt-2 mb-2"
+                size="small"
+                variant="text"
+                color="primary"
+                width="100%"
+            >
+                Reset options
+            </v-btn>
+            <v-divider></v-divider>
+        </div>
+    </v-expand-transition>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+
+const expand = ref(false);
+
+const location = ref('Before');
+
+const defaultMatchingOption = ref('Approximate');
+const currentMatchingOption = ref(defaultMatchingOption.value);
+const matchingOptions = ref(['Approximate', 'Exact']);
+
+const defaultSlicingOptions = ref('Substrings');
+const currentSlicingOption = ref(defaultSlicingOptions.value);
+const slicingOptions = ref(['Substrings', 'Spaces', 'Entire string']);
+
+const defaultSimilarityOptions = ref('None');
+const currentSimilarityOption = ref(defaultSimilarityOptions.value);
+const similarityOptions = ref(['None', 'Number to Letter', 'Letter to Number']);
+
+const defaultNumberOfMatches = ref(10000);
+const currentNumberOfMatches = ref(defaultNumberOfMatches.value);
+
+function reset() {
+    currentMatchingOption.value = defaultMatchingOption.value;
+    currentSlicingOption.value = defaultSlicingOptions.value;
+    currentSimilarityOption.value = defaultSimilarityOptions.value;
+    currentNumberOfMatches.value = defaultNumberOfMatches.value;
+}
+</script>
+
+<style lang="scss" scoped></style>
```

Comment:
The layout of the before-constraint and the after-constraint will always look the same (options etc). Wouldn't it be easier to manage both in one instead of having to copy everything?

value regex is similar as well.. Maybe there are some things that can be reused for each of them.

Reply:
Yeah sure will do that!